### ISSUE TITLE
Brightness not working - pi hub in strange state?

### ISSUE THREAD

**Issue by ancient-geek** [rricharz/pi-top-install#2]
Hi,

My pi-top running Rspbian seems stuck at a relatively low brightness level. If I run "brightness 9" to try to increase the brightness I see the following output but the brightness doesn't change:

brightness version 1.0
Sending: 0xFF
Receiving: 0xE4
Current brightness = 12
Requested brightness = 9
Sending: 0x48

I receive similar messages for brightness levels of 7,8,10 but if I run "brightness 6" the screen immediately dims to invisibility.

I note that the returned value for "current brightness"  of 12 seems out of the expected control range?

Can you suggest a solution to allow the brightness control to work ?

**Comment by rricharz** [comment]
Sorry,
I have no easy solution for this. It appears as if the hub-controller does not use the proper spi speed, because it does not accept the 0x48 sent, and the 0x4E received does not make sense. I have seen a very similar problem once when I tested another device on the spi bus, and it went away after leaving the pi-top turned off for a night. Have you done anything to the spi bus or installed any software which talks to that bus? Solutions would be to shutdown the pi-top for a while and/or to burn a second sd card with a fresh version of Raspbian to try. Please keep me informed.

**Comment by ancient-geek** [comment]
Hi Rene,

 

Thanks very much for the reply.

 

I had been experimenting with connecting an OELD up to the spi bus, so perhaps this upset the hub in some way.

 

I’ll try your suggestions and let you know the outcome.

 

Regards,

 

John

**Comment by rricharz** [comment]
I forgot to ask whether you have any software from pi-top installed. The last time I checked their hub-controller software on the rpi, it had a bug which would prevent it from recovering from such a situation. It opened the connection to the spi bus once at bootup, and then talked every 250 msec to the bus, assuming that nothing else had changed the bus speed or anything else relevant on the spi bus, and it kept sending bad data to the spi bus if it had received bad data. I don't know whether their most recent version still has that problem.

**Comment by ancient-geek** [comment]
Hi Rene,

 

Following your clue that the hub spi had got into an unusual state I thought it was worth trying to force the hub micro-controller to reboot and hence get back into a known state.

 

So, I disconnected the pi-top from external power and unscrewed and lifted the hub PCB to disconnect it from the pi-top battery for 30 seconds.

 

I then located and screwed the hub back in the pi-top, started the pi-top and voila! the brightness control and shutdown are working properly again.

 

Note, I did not need to use a fresh install of Raspbian.

 

Conclusion: one needs to be careful experimenting with spi as it is clearly possible to put the hub into a spurious state. However, if this happens disconnecting the hub from external power and the battery seems to fix things. 

 

Thanks very much for your help and hope this might be helpful to others.

 

Regards,

 

John
